File: python/easy/reverse_string_ii.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        length = len(s)
        ret = ""
        start = 0
        end = k-1
        while(1):
            if length <= k:
                ret += s[start:length][::-1]
                break
            ret += s[start:end+1][::-1]
            ret += s[end+1:end+k+1]
            if start+2*k < length-1 and end+2*k >= length-1:
                ret += s[(start+2*k):length][::-1]
                break
            elif start+2*k >= length:
                break
            else:
                start += 2*k
                end += 2*k
        return ret

File: cpp/easy/reverse_string_ii.cpp
 class Solution {
public:
    string reverseStr(string s, int k) {
        if(s.size() == 0 || s.size() == 1)
            return s;
        int start = 0;
        int end = k-1;
        while(1)
        {
            //check whether or not the input string size is less than integer k
            if(s.size() <= k)
            {
                reverse_string(s, start, s.size()-1);
                break;
            }
            reverse_string(s, start, end);
            if(start+2*k < s.size()-1 && end+2*k > s.size()-1)
            {
                reverse_string(s, start+2*k, s.size()-1);
                break;
            }
            else if(start+2*k >= s.size()-1)
                break;
            else{
                start = start + (2*k);
                end = end + (2*k);
            }
            
        }
        return s;
    }
    
    void reverse_string(string& s, int start, int end)
    {
        if(start >= end)
            return;
        while(start < end)
        {
            char tmp = s[start];
            s[start] = s[end];
            s[end] = tmp;
            ++start;
            --end;
        }
    }
};

Conclusion:
easy
